bootloader/firmware/stm32L5/node_id_stm32l5xx.c

```diff
@@ -5,92 +5,18 @@
 
 #if defined(node_id_pipette_dynamic)
 
-#define ADC_MAX_VAL 4095
-#define ADC_REF_MV  3300
 static CANNodeId dynamic_id_backing = can_nodeid_broadcast;
-static ADC_HandleTypeDef hadc;
-
-static bool MX_ADC1_Init(void) {
-    /** Common config
-     */
-    hadc.Instance = ADC1;
-    hadc.Init.ClockPrescaler = ADC_CLOCK_SYNC_PCLK_DIV4;
-    hadc.Init.Resolution = ADC_RESOLUTION_12B;
-    hadc.Init.DataAlign = ADC_DATAALIGN_RIGHT;
-    hadc.Init.ScanConvMode = ADC_SCAN_DISABLE;
-    hadc.Init.EOCSelection = ADC_EOC_SINGLE_CONV;
-    hadc.Init.LowPowerAutoWait = DISABLE;
-    hadc.Init.ContinuousConvMode = DISABLE;
-    hadc.Init.NbrOfConversion = 1;
-    hadc.Init.DiscontinuousConvMode = DISABLE;
-    hadc.Init.ExternalTrigConv = ADC_SOFTWARE_START;
-    hadc.Init.ExternalTrigConvEdge = ADC_EXTERNALTRIGCONVEDGE_NONE;
-    hadc.Init.DMAContinuousRequests = DISABLE;
-    hadc.Init.Overrun = ADC_OVR_DATA_PRESERVED;
-    hadc.Init.OversamplingMode = DISABLE;
-    return HAL_OK == HAL_ADC_Init(&hadc);
-}
-
-
-/**
- * @brief ADC MSP Initialization
- * This function configures the hardware resources used in this example
- * @param hadc: ADC handle pointer
- * @retval None
- */
-void HAL_ADC_MspInit(ADC_HandleTypeDef* which_adc) {
-    GPIO_InitTypeDef GPIO_InitStruct = {0};
-    if (which_adc->Instance == ADC1) {
-        /* Peripheral clock enable */
-        __HAL_RCC_ADC_CLK_ENABLE();
-
-        __HAL_RCC_GPIOB_CLK_ENABLE();
-
-        GPIO_InitStruct.Pin = GPIO_PIN_1;
-        GPIO_InitStruct.Mode = GPIO_MODE_ANALOG;
-        GPIO_InitStruct.Pull = GPIO_NOPULL;
-        HAL_GPIO_Init(GPIOB, &GPIO_InitStruct);
-    }
-}
-
-static bool adc_init(void) {
-    if (!MX_ADC1_Init()) {
-        return false;
-    }
-    if (HAL_ADCEx_Calibration_Start(&hadc, ADC_SINGLE_ENDED) != HAL_OK) {
-        /* Calibration Error */
-        return false;
-    }
-    ADC_ChannelConfTypeDef sConfig = {0};
-    // Configure channel 16 (PB1) for single ended long duration read on
-    // the tool ID pin
-    sConfig.Channel = ADC_CHANNEL_16;
-    sConfig.Rank = ADC_REGULAR_RANK_1;
-    sConfig.SamplingTime = ADC_SAMPLETIME_640CYCLES_5;
-    sConfig.SingleDiff = ADC_SINGLE_ENDED;
-    sConfig.OffsetNumber = ADC_OFFSET_NONE;
-    sConfig.Offset = 0;
-    return HAL_OK == HAL_ADC_ConfigChannel(&hadc, &sConfig);
-}
-
-
-static uint16_t adc_read(void) {
-    HAL_ADC_Start(&hadc);
-    HAL_ADC_PollForConversion(&hadc, HAL_MAX_DELAY);
-    return HAL_ADC_GetValue(&hadc);
-}
-
-static uint16_t adc_convert(uint16_t adc_raw_reading) {
-    return ((((uint32_t)adc_raw_reading) * ADC_REF_MV) / ADC_MAX_VAL);
-}
-
 
 static CANNodeId update_dynamic_nodeid() {
-    if (!adc_init()) {
-        return can_nodeid_pipette_left_bootloader;
-    }
+    // B1: mount id
+    __HAL_RCC_GPIOB_CLK_ENABLE();
```

Comment:
I don't even know if I would bother changing this file. Pre-EVT pipettes aren't compatible with today's robot, so the likelihood that they would be attached to an EVT/DVT head board is pretty much zero.